src/js/services/app-service.js

```diff
@@ -0,0 +1,34 @@
+import { store } from 'stores/app-store';
+import * as AppStateActions from 'actions/app-state-actions';
+import { observable } from 'utils/events';
+
+
+export function openWidget() {
+    let {embedded} = store.getState().appState;
+    if (!embedded) {
+        store.dispatch(AppStateActions.openWidget());
+        observable.trigger('widget:opened');
+    }
+}
+
+export function closeWidget() {
+    let {embedded} = store.getState().appState;
+    if (!embedded) {
+        store.dispatch(AppStateActions.closeWidget());
+        observable.trigger('widget:closed');
+    }
+}
+
+
+export function toggleWidget() {
+    let {embedded, widgetOpened} = store.getState().appState;
+    if (!embedded) {
+        store.dispatch(AppStateActions.toggleWidget());
```

Comment:
They have just as many side effects as this function does, no? Those functions are dispatching actions to the store, and triggering events on the `observable` like this one is doing

```
export function toggleWidget() {
    let {embedded, widgetOpened} = store.getState().appState;
    if (!embedded) {
        if (widgetOpened) {
            closeWidget();
        } else {
            openWidget();
        }
    }
}
```